miden-tx/src/errors/mod.rs

```diff
@@ -43,18 +43,23 @@ impl std::error::Error for TransactionExecutorError {}
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum TransactionProverError {
-    ProveTransactionProgramFailed(ExecutionError),
+    DeserializationError(DeserializationError),
+    InternalError(String),
```

Comment:
nit: I'm not sure about having a separate `DeserializationError` variant as it is very specific to the remote TX prover. Maybe we combine these two and have just an `InternalError` or `OtherError` or something like that?

Reply:
Changed this and the renamed `ProveTransactionProgramFailed` to `TransactionProgramExecutionFailed`!